Library/ViewModels/PledgeViewModelTests.swift

```diff
@@ -2602,6 +2602,110 @@ final class PledgeViewModelTests: TestCase {
+        .template
+          |> Backing.lens.paymentSource .~ Backing.PaymentSource.template
+          |> Backing.lens.status .~ .pledged
+          |> Backing.lens.reward .~ reward
+          |> Backing.lens.rewardId .~ reward.id
+          |> Backing.lens.shippingAmount .~ 10
+          |> Backing.lens.amount .~ 700.0
+      )
+
+    let updateBackingEnvelope = UpdateBackingEnvelope(
+      updateBacking: .init(
+        checkout: .init(
+          id: "Q2hlY2tvdXQtMQ==",
+          state: .successful,
+          backing: .init(
+            clientSecret: "client-secret",
+            requiresAction: false
+          )
+        )
+      )
+    )
+
+    let mockOptimizelyClient = MockOptimizelyClient()
+      |> \.experiments
+      .~ [
+        OptimizelyExperiment.Key.nativeRiskMessaging.rawValue:
+          OptimizelyExperiment.Variant.variant1.rawValue
+      ]
+
+    let mockService = MockService(
+      updateBackingResult: .success(updateBackingEnvelope)
+    )
+
+    withEnvironment(apiService: mockService, currentUser: .template, optimizelyClient: mockOptimizelyClient) {
+      let data = PledgeViewData(
+        project: project,
+        rewards: [reward],
+        selectedQuantities: [reward.id: 1],
+        selectedLocationId: nil,
+        refTag: .discovery,
+        context: .update
+      )
+
+      self.vm.inputs.configure(with: data)
+      self.vm.inputs.viewDidLoad()
+
+      self.beginSCAFlowWithClientSecret.assertDidNotEmitValue()
+      self.notifyDelegateUpdatePledgeDidSucceedWithMessage.assertDidNotEmitValue()
+      self.popToRootViewController.assertDidNotEmitValue()
+      self.showErrorBannerWithMessage.assertDidNotEmitValue()
+      self.configurePledgeViewCTAContainerViewIsEnabled.assertValues([false])
+      self.processingViewIsHidden.assertDidNotEmitValue()
+
+      self.vm.inputs.pledgeAmountViewControllerDidUpdate(
+        with: (amount: 25.0, min: 25.0, max: 10_000.0, isValid: true)
+      )
+
+      self.beginSCAFlowWithClientSecret.assertDidNotEmitValue()
+      self.notifyDelegateUpdatePledgeDidSucceedWithMessage.assertDidNotEmitValue()
+      self.popToRootViewController.assertDidNotEmitValue()
+      self.showErrorBannerWithMessage.assertDidNotEmitValue()
+      self.configurePledgeViewCTAContainerViewIsEnabled.assertValues([false, true])
+      self.processingViewIsHidden.assertDidNotEmitValue()
+
+      self.vm.inputs.shippingRuleSelected(.template)
+
+      self.beginSCAFlowWithClientSecret.assertDidNotEmitValue()
+      self.notifyDelegateUpdatePledgeDidSucceedWithMessage.assertDidNotEmitValue()
+      self.popToRootViewController.assertDidNotEmitValue()
+      self.showErrorBannerWithMessage.assertDidNotEmitValue()
+      self.configurePledgeViewCTAContainerViewIsEnabled.assertValues([false, true])
+      self.processingViewIsHidden.assertDidNotEmitValue()
+
+      self.vm.inputs.submitButtonTapped()
+
+      self.goToRiskMessagingModal.assertDidNotEmitValue()
```

Comment:
This is the only output we are testing, so to reduce bloat in this test, the entire test after `self.vm.inputs.viewDidLoad()` could be 
```
      self.vm.inputs.pledgeAmountViewControllerDidUpdate(
        with: (amount: 25.0, min: 25.0, max: 10_000.0, isValid: true)
      )
      self.vm.inputs.shippingRuleSelected(.template)
      self.vm.inputs.submitButtonTapped()
      self.goToRiskMessagingModal.assertDidNotEmitValue()
```

Because that is all we really care about in this particular patch. Also then the test could be more specific to that use `testRiskModalNotShowing_WhenUpdating_Success`

Unless you want to test everything again under these specific conditions, which is also valid. Just a thought.